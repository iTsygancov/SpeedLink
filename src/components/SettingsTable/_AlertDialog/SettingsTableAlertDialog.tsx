import {
  AlertDialog,
  AlertDialogAction,
  AlertDialogCancel,
  AlertDialogContent,
  AlertDialogDescription,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogTitle
} from "@/components/ui/alert-dialog";
import { Shortcut } from "@/types";

export type SettingsTableAlertDialogProps = {
  currentShortcut: Shortcut;
  initialShortcut: Shortcut;
  isDialogOpen: boolean;
  handleDeleteShortcut: (id: string) => void;
  setIsDialogOpen: (isDialogOpen: boolean) => void;
  setCurrentShortcut: (item: Shortcut) => void;
};

const SettingsTableAlertDialog = ({
  currentShortcut,
  initialShortcut,
  isDialogOpen,
  handleDeleteShortcut,
  setIsDialogOpen,
  setCurrentShortcut
}: SettingsTableAlertDialogProps) => {
  const handleCancelClick = () => {
    setCurrentShortcut(initialShortcut);
    setIsDialogOpen(false);
  };

  const handleDeleteClick = () => {
    handleDeleteShortcut(currentShortcut.id);
  };

  return (
    <AlertDialog open={isDialogOpen}>
      <AlertDialogContent>
        <AlertDialogHeader>
          <AlertDialogTitle>Are you absolutely sure?</AlertDialogTitle>
          <AlertDialogDescription>
            This action cannot be undone. This will permanently delete your
            shortcut.
          </AlertDialogDescription>
        </AlertDialogHeader>
        <AlertDialogFooter>
          <AlertDialogCancel onClick={handleCancelClick}>
            Cancel
          </AlertDialogCancel>
          <AlertDialogAction onClick={handleDeleteClick}>
            Continue
          </AlertDialogAction>
        </AlertDialogFooter>
      </AlertDialogContent>
    </AlertDialog>
  );
};

export default SettingsTableAlertDialog;
