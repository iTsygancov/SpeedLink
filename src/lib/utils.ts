import { type ClassValue, clsx } from "clsx";
import { twMerge } from "tailwind-merge";

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}

export function generateAlphanumericKeysArray() {
  const keysArray = [];

  // Generate lowercase alphabetic keys
  for (let i = 65; i <= 90; i++) {
    keysArray.push(String.fromCharCode(i));
  }

  // Generate numeric keys
  for (let i = 48; i <= 57; i++) {
    keysArray.push(String.fromCharCode(i));
  }

  return keysArray;
}
