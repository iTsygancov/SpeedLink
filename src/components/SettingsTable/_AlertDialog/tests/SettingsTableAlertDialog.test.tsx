import SettingsTableAlertDialog, {
  SettingsTableAlertDialogProps
} from "../SettingsTableAlertDialog";
import { initialProps } from "./SettingsTableAlertDialog.data";
import { expectToMatchSnapshot, renderTree } from "@/test/utils";
import { describe, it } from "vitest";

const testComponentWithProps = (props: SettingsTableAlertDialogProps) => {
  expectToMatchSnapshot(renderTree(<SettingsTableAlertDialog {...props} />));
};

describe("SettingsTableAlertDialog", () => {
  it("should render null", () => {
    testComponentWithProps(initialProps);
  });
});
