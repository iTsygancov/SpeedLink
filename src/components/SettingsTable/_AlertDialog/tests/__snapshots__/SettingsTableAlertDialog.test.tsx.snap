// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsTableAlertDialog > should render null 1`] = `null`;
